# HG changeset patch
# User rk
# Date 1285925928 -7200
# Node ID 7f70824841b6bbbc0dd8a4351b420acce53eeacb
# Parent  5ff48408d5fedc61b885a44c8b5b0b72bafe643e
wxDataViewCtrl::Expand() needs to expand all ancestors of the given item before expanding the item itself

diff -r 5ff48408d5fe -r 7f70824841b6 src/generic/datavgen.cpp
--- a/src/generic/datavgen.cpp	Fri Oct 01 11:37:01 2010 +0200
+++ b/src/generic/datavgen.cpp	Fri Oct 01 11:38:48 2010 +0200
@@ -4416,6 +4416,8 @@ int wxDataViewCtrl::GetRowByItem( const 
 
 void wxDataViewCtrl::Expand( const wxDataViewItem & item )
 {
+    ExpandAncestors( item );
+
     int row = m_clientArea->GetRowByItem( item );
     if (row != -1)
         m_clientArea->Expand(row);
